### PR TITLE
Update eslint plugins

### PR DESCRIPTION
**Pull request by TylerHorth**
@GoodForOneFare @lemonmade

### CODE REVIEW

File: packages/eslint-plugin-shopify/lib/config/rules/babel.js [hunk 1]
```diff
@@ -2,22 +2,12 @@
...
-  // Handles destructuring arrays with flow type in function parameters
-  'babel/array-bracket-spacing': ['warn', 'never'],
-  // Handles async functions correctly
-  'babel/arrow-parens': ['warn', 'always'],
-  // Require a particular separator between properties in Flow object types.
-  'babel/flow-object-type': ['warn', 'comma'],
-  // Require or forbid trailing commas for function paramater lists.
-  'babel/func-params-comma-dangle': ['warn', 'always-multiline'],
-  // Handles async/await functions correctly
-  'babel/generator-star-spacing': ['warn', 'after'],
```
Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/babel.js]: Can you move all this configuration stuff to the matching non-Babel rules? I think we turned off all of these from core ESLint when we needed this plugin to lint them with some newer ES features.

File: packages/eslint-plugin-shopify/lib/config/rules/flowtype.js [hunk 2]
```diff
@@ -11,18 +11,24 @@ module.exports = {
...
+  // Disallows use of primitive constructors as types, such as Boolean, Number and String.
+  'flowtype/no-primitive-constructor-types': 'error',
   // Warns against weak type annotations any, Object and Function.
   'flowtype/no-weak-types': ['error', {
     any: false,
     Object: false,
     Function: true,
   }],
+  // Enforces consistent separators between properties in Flow object types.
+  'flowtype/object-type-delimiter': 'warn',
```
Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/flowtype.js]: Can you explicitly set this to `['warn', 'comma']`?

**Comment by TylerHorth** [comment]
@lemonmade How's this look now? 🙃

File: packages/eslint-plugin-shopify/lib/config/rules/babel.js [hunk 3]
```diff
@@ -2,22 +2,12 @@
...
 
 module.exports = {
-  // Handles destructuring arrays with flow type in function parameters
-  'babel/array-bracket-spacing': ['warn', 'never'],
-  // Handles async functions correctly
-  'babel/arrow-parens': ['warn', 'always'],
-  // Require a particular separator between properties in Flow object types.
-  'babel/flow-object-type': ['warn', 'comma'],
-  // Require or forbid trailing commas for function paramater lists.
-  'babel/func-params-comma-dangle': ['warn', 'always-multiline'],
```
Comment (lemonmade) [inline packages/eslint-plugin-shopify/lib/config/rules/babel.js]: Just for my own sanity, this rule is still applied, but is done by the default `comma-dangle` rule, yes?

**Comment by lemonmade** [comment]
One question, otherwise looks great 👍

File: packages/eslint-plugin-shopify/lib/config/rules/babel.js (hunk 3 above)
Comment (TylerHorth) [inline packages/eslint-plugin-shopify/lib/config/rules/babel.js]: Correct! 👍

**Pull request merged**